packages/ui-components/src/FormDialog.tsx

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright 2018-2020 IBM Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import '../style/index.css';
+
+import { Dialog } from '@jupyterlab/apputils';
+import { Widget } from '@lumino/widgets';
+
+const DEFAULT_BUTTON_CLASS = 'elyra-DialogDefaultButton';
+/*
+ * Validate required dialog fields upon display
+ * - Provides a generic validation by checking if required form fields are populated
+ * - Expect required fields in dialog body to contain attribute: data-form-required
+ * - NOTE: The Dialog widget will skip any validation upon pressing 'enter' key and resolve the dialog as is
```

Comment:
you may be able to capture the `keydown` event and prevent propagation/default behavior if **Enter** key is pressed and all required are not filled

Reply:
Thank you for the directions, overriding the dialog eventHandler worked for this case :)